include/nbl/system/CMemoryFile.h

```diff
@@ -0,0 +1,54 @@
+#ifndef C_MEMORY_FILE_H
+#define C_MEMORY_FILE_H
+#include <nbl/system/IFile.h>
+
+namespace nbl::system
+{
+class CMemoryFile : public IFile
+{
+public:
+	CMemoryFile(const std::filesystem::path& _name, std::underlying_type_t<E_CREATE_FLAGS> _flags) : IFile(_flags), m_name(_name)
+	{
+
+	}
+
+	virtual const std::filesystem::path& getFileName() const override
+	{
+		return m_name;
+	}
+
+	virtual void* getMappedPointer() override { return nullptr; }
+	virtual const void* getMappedPointer() const override { return nullptr; }
```

Comment:
make it return m_buffer.data()
and force constructor adding MAPPABLE and COHERENT to flags

Reply:
can you guys call it a `CFileView` instead of memroy file?

Reply:
kk